### ISSUE TITLE
Unavailable Private Videos Crash Rehike

### ISSUE THREAD

**Issue by 9p4** [Rehike/Rehike#611]
### Rehike version

0.8.3 / nightly 90cfa98

### Operating system

Rocky Linux 9.4 5.14.0-427.24.1.el9_4.x86_64

### PHP version

8.3.11 php-fpm (Remi)

### What's going on?

When attempting to view a private video in Rehike, it crashes. For example: https://www.youtube.com/watch?v=BY-TYGpiGpU

Crash log: [20240921182228-watch.txt](https://github.com/user-attachments/files/17085899/20240921182228-watch.txt)

Traceback:

```
TypeError: Rehike\Model\Watch\WatchBakery::bakeSecondaryResults(): Return value must be of type object, null returned
  at Rehike\Model\Watch\WatchBakery::bakeSecondaryResults()(rehike://models/Watch/WatchBakery.php:400)
  at Rehike\Model\Watch\WatchBakery::Rehike\Model\Watch\{closure}()(rehike://models/Watch/WatchBakery.php:151)
  at Generator::send()({unknown}:)
  at Rehike\Async\Concurrency\AsyncFunction::Rehike\Async\Concurrency\{closure}()(rehike://modules/Rehike/Async/Concurrency/AsyncFunction.php:133)
  at YukisCoffee\CoffeeRequest\Promise::then()(rehike://modules/YukisCoffee/CoffeeRequest/Promise.php:196)
  at Rehike\Async\Concurrency\AsyncFunction::run()(rehike://modules/Rehike/Async/Concurrency/AsyncFunction.php:113)
  at Rehike\Async\Concurrency::async()(rehike://modules/Rehike/Async/Concurrency.php:138)
  at Rehike\Async\async()(rehike://includes/functions/async.php:91)
  at Rehike\Model\Watch\WatchBakery::bake()(rehike://models/Watch/WatchBakery.php:86)
  at Rehike\Controller\core\NirvanaController@anonymous(#1)<rehike://controllers/watch.php:29>::{closure}()(rehike://controllers/watch.php:266)
  at YukisCoffee\CoffeeRequest\Promise::resolve()(rehike://modules/YukisCoffee/CoffeeRequest/Promise.php:285)
  at YukisCoffee\CoffeeRequest\Util\PromiseAllBase::resolve()(rehike://modules/YukisCoffee/CoffeeRequest/Deferred.php:54)
  at YukisCoffee\CoffeeRequest\Util\PromiseAllBase::handlePromiseResolution()(rehike://modules/YukisCoffee/CoffeeRequest/Util/PromiseAllBase.php:70)
  at YukisCoffee\CoffeeRequest\Util\PromiseAllBase::YukisCoffee\CoffeeRequest\Util\{closure}()(rehike://modules/YukisCoffee/CoffeeRequest/Util/PromiseAllBase.php:127)
  at YukisCoffee\CoffeeRequest\Promise::resolve()(rehike://modules/YukisCoffee/CoffeeRequest/Promise.php:285)
  at YukisCoffee\CoffeeRequest\Util\QueuedPromiseResolver::finish()(rehike://modules/YukisCoffee/CoffeeRequest/Util/QueuedPromiseResolver.php:57)
  at YukisCoffee\CoffeeRequest\Loop::finishQueuedPromises()(rehike://modules/YukisCoffee/CoffeeRequest/Loop.php:240)
  at YukisCoffee\CoffeeRequest\Loop::cleanup()(rehike://modules/YukisCoffee/CoffeeRequest/Loop.php:230)
  at YukisCoffee\CoffeeRequest\Loop::run()(rehike://modules/YukisCoffee/CoffeeRequest/Loop.php:84)
  at YukisCoffee\CoffeeRequest\CoffeeRequest::run()(rehike://modules/YukisCoffee/CoffeeRequest/CoffeeRequest.php:119)
  at Rehike\Network::run()(rehike://modules/Rehike/Network.php:373)
  at Rehike\Controller\core\HitchhikerController::get()(rehike://controllers/core/HitchhikerController.php:107)
  at Rehike\ControllerV2\GetControllerInstance::_cv2WrappedCallControllerMethod()(rehike://modules/Rehike/ControllerV2/GetControllerInstance.php:51)
  at Rehike\ControllerV2\GetControllerInstance::__call()(rehike://modules/Rehike/ControllerV2/GetControllerInstance.php:74)
  at Rehike\ControllerV2\Router::pointerHandler()(rehike://modules/Rehike/ControllerV2/Router.php:253)
  at Rehike\ControllerV2\Router::baseRequestMethod()(rehike://modules/Rehike/ControllerV2/Router.php:192)
  at Rehike\ControllerV2\Router::get()(rehike://modules/Rehike/ControllerV2/Router.php:40)
  at require("/var/www/youtube.com/router.php")(rehike://router.php:81)
  at Rehike\Boot\Bootloader::postboot()(rehike://modules/Rehike/Boot/Bootloader.php:134)
  at Rehike\Boot\Bootloader::startSession()(rehike://modules/Rehike/Boot/Bootloader.php:28)
  at {main}(rehike://index.php:73)
```

Message logs:
```
[{"host":"www.youtube.com","ttl":300,"class":"IN","type":"CNAME","target":"youtube-ui.l.google.com"}]
youtube-ui.l.google.com [{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.182.190"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.177.91"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.176.136"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.178.93"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.182.136"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.177.190"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.178.136"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.4.136"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.178.91"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.176.190"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.178.190"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.176.93"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.182.91"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.176.91"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.182.93"},{"host":"youtube-ui.l.google.com","ttl":83,"class":"IN","type":"A","ip":"142.251.177.93"}]
```

**Comment by 9p4** [comment]
The marked commit fixes the crash, but it does return "Content warning: This video may be inappropriate for some users." instead of "This video is private".

**Comment by kawapure** [comment]
That problem is now fixed: 4ba7520

**Comment by 9p4** [comment]
Thank you so much!
